daemon/src/wire.rs

```diff
@@ -20,13 +20,15 @@ use tokio_util::codec::{Decoder, Encoder, LengthDelimitedCodec};
 #[allow(clippy::large_enum_variant)]
 pub enum TakerToMaker {
     TakeOrder { order_id: OrderId, quantity: Usd },
+    ProposeSettlement(SettlementProposal),
```

Comment:
Instead of embedding this, can we just have fields in the enum? Would like to not couple the domain model too much to wire message format!